### PR TITLE
Image refresh for ubuntu-stable [no-test]

### PR DESCRIPTION
**Pull request by cockpituous**
Image refresh for ubuntu-stable
 * [x] image-refresh ubuntu-stable

### CODE REVIEW

**Comment by marusak** [comment]
`image-prepare` fails constantly, this needs some investigation

**Comment by KKoukiou** [comment]
> `image-prepare` fails constantly, this needs some investigation

It's because firewalld is not running, 
```
Created symlink /etc/systemd/system/sockets.target.wants/cockpit.socket -> /lib/systemd/system/cockpit.socket.
+ systemctl disable cockpit.socket
Removed /etc/systemd/system/sockets.target.wants/cockpit.socket.
+ systemctl disable tuned.service
+ firewall-cmd --add-service=cockpit --permanent
FirewallD is not running
Traceback (most recent call last):
  File "bots/image-prepare", line 206, in <module>
    sys.exit(main())
  File "bots/image-prepare", line 88, in main
    build_and_install(build_image, results, skips, args)
  File "bots/image-prepare", line 165, in build_and_install
    run_install_script(machine, True, True, skips, os.path.basename(source), args)
  File "bots/image-prepare", line 139, in run_install_script
    machine.execute(cmd, timeout=1800)
  File "/home/kkoukiou/repos/cockpit/bots/machine/machine_core/ssh_connection.py", line 346, in execute
    raise subprocess.CalledProcessError(proc.returncode, command, output=output)
subprocess.CalledProcessError: Command 'cd /var/tmp; ./ubuntu-stable.install --build --install --skip 'cockpit-ostree' 'cockpit-192.x.tar.gz'' returned non-zero exit status 252.

```

**Comment by KKoukiou** [comment]
Did some debugging, why the firewall is not starting, looks like some iptables related packages ar e missing. Here is the journal related messages:
```
root@ibm-p8-kvm-03-guest-02:~# journalctl -b | grep firewall
Apr 23 18:54:50 ibm-p8-kvm-03-guest-02 systemd[1]: Starting firewalld - dynamic firewall daemon...
Apr 23 18:54:51 ibm-p8-kvm-03-guest-02 systemd[1]: Started firewalld - dynamic firewall daemon.
Apr 23 18:54:51 ibm-p8-kvm-03-guest-02 firewalld[535]: WARNING: iptables-restore and iptables are missing, disabling IPv4 firewall.
Apr 23 18:54:51 ibm-p8-kvm-03-guest-02 firewalld[535]: WARNING: ip6tables-restore and ip6tables are missing, disabling IPv6 firewall.
Apr 23 18:54:52 ibm-p8-kvm-03-guest-02 firewalld[535]: FATAL ERROR: No IPv4 and IPv6 firewall.
Apr 23 18:54:52 ibm-p8-kvm-03-guest-02 firewalld[535]: ERROR: Raising SystemExit in run_server
Apr 23 18:54:52 ibm-p8-kvm-03-guest-02 systemd[1]: firewalld.service: Succeeded.
Apr 23 18:54:53 ibm-p8-kvm-03-guest-02 NetworkManager[1130]: <warn>  [1556060093.0029] firewall: [0x7ff5600068f0,change:"eth1"]: complete: request failed (Message recipient disconnected from message bus without replying)
```

**Comment by martinpitt** [comment]
@KKoukiou : Ubuntu 19.04 just got released, so this moves the ubuntu-stable image from 18.10 to 19.04. So some larger changes are to be expected. This sounds like a missing package dependency somewhere, thus something to report downstream. I can have a look tomorrow.

**Comment by martinpitt** [comment]
The image does have ip{,6}tables{,-restore}, and also "ebtables". There are [two](https://bugs.launchpad.net/ubuntu/+source/firewalld/+bug/1821596) [recent](https://bugs.launchpad.net/ubuntu/+source/firewalld/+bug/1799095) bug reports, but they don't match this error.

According to the second bug, the default backend should be iptables, not nftables; this matches /etc/firewalld/firewalld.conf. When I switch it to nftables, firewalld at least starts, but also shows a lot of

    WARNING: COMMAND_FAILED: UNKNOWN_ERROR: 'ip4tables' backend does not exist

I reported this downstream: https://bugs.launchpad.net/ubuntu/+source/firewalld/+bug/1826187

**Comment by martinpitt** [comment]
I pushed a workaround for the above. There are now still some failures at least in the firewalld tests (so keeping "needswork"), but let's get an initial full test run.

**Comment by marusak** [comment]
Uh, 32 failed tests, this will need some work :)
I just scrolled through the issues and here are some quick notes:

- packagekit has issues - majority of tests dealing with packagekit failed
- docker has issues (`systemctl start docker || systemctl start docker-latest` failes)
- libvirtd is coredumping
- firewall tests fail (kinda expected?)

Others may be just flakes, or maybe some new regressions. But solving these 4 issues could get us down to a reasonable number.

**Comment by martinpitt** [comment]
I pushed fixes for the packagekit and firewalld test failures.

**Comment by marusak** [comment]
Docker cannot start also due to firewall stuff
`dockerd[6498]: Error starting daemon: Error initializing network controller: error obtaining controller instance: failed to create NAT chain DOCKER: COMMAND_FAILED: UNKNOWN_ERROR: 'ip4tables' backend does not exist`
I will continue thinking tomorrow if we can do something about it.
And for the libvirtd coredumping I lost url of previous run, so I need to wait for another run to see it again.

**Comment by marusak** [comment]
@martinpitt I think that switching to `nftables` was not the best move as it broke some things. I check why it states that that ip{,6}tables{,-restore} are missing even though they are present and I think it is looking for it at a wrong path. I tried to symlink these commands and it worked for me. I pushed it here (effectively rewriting you commit) to see if it really helps.

File: bots/images/scripts/ubuntu-stable.install [hunk 1]
```diff
@@ -2,6 +2,13 @@
 
 set -e
 
+# HACK: TODO file some issue for this
```
Comment (martinpitt) [inline bots/images/scripts/ubuntu-stable.install]: We already have the downstream issue: https://launchpad.net/bugs/1826187 . I followed up there to point out that workaround/root cause.

**Comment by martinpitt** [comment]
@marusak: Nice! if that works, it's a much better workaround indeed! And that also gives an useful explanation what's actually wrong. I followed up to the downstream Launchpad bug report.

**Comment by martinpitt** [comment]
This looks much better! I can reproduce the testCreateTimers failure, on shutdown there's an eternal hang of pcp:

    [  *** ] A stop job is running for Performan…rchive Logger (2min 42s / 3min 1s)

I can also reproduce testResizeLuks, and the check-realms unexpected tuned message needs investigating as well.

I updated the TODO list above accordingly.

**Comment by martinpitt** [comment]
I pushed another update with the storage-resize fix, and squashing/cleaning up the firewalld workaround.

**Comment by martinpitt** [comment]
The pcp shutdown hang also happens on [fedora](https://209.132.184.41:8493/logs/pull-11618-20190425-071717-58dd9672-verify-fedora-30/log.html), so that's nothing new. But there it eventually stops. This sounds related to https://bugzilla.redhat.com/show_bug.cgi?id=1188193 , a very old bug.

We could just increase the reboot timeout more, but this starts being ridiculous. I'll think about another  workaround instead.

**Comment by martinpitt** [comment]
Pushed a workaround for the reboot hang and another flake fix from the [previous test run](https://209.132.184.41:8493/logs/pull-11651-20190426-062626-0aae5e66-verify-ubuntu-stable/log.html).

**Comment by martinpitt** [comment]
I investigated and [reported](https://bugzilla.redhat.com/show_bug.cgi?id=1703348) the PCP hang, and updated the bug reference in the commit log.

**Comment by martinpitt** [comment]
Next [two](https://209.132.184.41:8493/logs/pull-11651-20190426-073939-3a54dd46-verify-ubuntu-stable/log.html) [test runs](https://209.132.184.41:8493/logs/pull-11651-20190426-075959-f7f9a93a-verify-ubuntu-stable/log.html) are again very flaky, retrying.

**Comment by martinpitt** [comment]
[so close](https://209.132.184.41:8493/logs/pull-11651-20190426-100303-f7f9a93a-verify-ubuntu-stable/log.html)!

**Pull request merged**